nodejs/awsx/ec2/vpc.ts

```diff
@@ -49,18 +50,47 @@ export class Vpc extends pulumi.ComponentResource {
      */
     public readonly natGateways: x.ec2.NatGateway[] = [];
 
-    constructor(name: string, args?: VpcArgs, opts?: pulumi.ComponentResourceOptions);
-    constructor(name: string, args?: ExistingVpcArgs, opts?: pulumi.ComponentResourceOptions);
-    constructor(name: string, args: VpcArgs | ExistingVpcArgs = {}, opts: pulumi.ComponentResourceOptions = {}) {
-        super("awsx:x:ec2:Vpc", name, {}, opts);
+    private readonly parent: Vpc;
+
+    constructor(name: string, parent: Vpc,
+                args: InternalVpcArgs | ExistingVpcArgs | ExistingVpcIdArgs,
+                opts: pulumi.ComponentResourceOptions = {}) {
```

Comment:
the code change here is that we can be totally synchronous in Vpc data because we pass in `InternalVpcArgs` (not `VpcArgs`).  The former has all data we need realized.